### PR TITLE
feat: add `Map.{maximum*}`

### LINKED ISSUE #2980: Introduce `Map.{maximumKey, maximumKeyBy, maximumValue, maximumValueBy}`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #2980

### CODE REVIEW

File: main/src/library/RedBlackTree.flix [hunk 1]
```diff
@@ -679,53 +679,76 @@ namespace RedBlackTree {
...
 
     ///
     /// Applies `cmp` over the tree `t` sequentially and optionally returns the lowest
     /// element according to `cmp`.
     ///
-    /// Note that this is equivalent to `seqMinimumBy` but for performance reasons -- to avoid
-    /// megamorphic calls -- we use a copy here.
-    ///
-    def parSeqMinimumBy(cmp: (k, v, k, v) -> Comparison & ef, t: RedBlackTree[k, v]): Option[(k, v)] & ef = match t {
+    def parSeqLimitBy(decider: Comparison, cmp: (k, v, k, v) -> Comparison & ef, t: RedBlackTree[k, v]): Option[(k, v)] & ef = match t {
```
Comment (magnus-madsen) [inline main/src/library/RedBlackTree.flix:738]: I think the role of decider is to know if you are looking for the smallest/largest element. And then `cmp` is used for the comparison. My question is: Could we not have a function like `(k, v, k, v) -> (k, v)` that simply returns the right element (i.e. either the smallest or largest)?

Comment (jaschdoc) [reply main/src/library/RedBlackTree.flix:738]: Something like `...f(cmp(kl, vl, kr, vr))...`, `f: Comparison -> (k, v)`?

Comment (jaschdoc) [reply main/src/library/RedBlackTree.flix:738]: I think I understand what you mean. I'll push a draft in a bit

Comment (magnus-madsen) [reply main/src/library/RedBlackTree.flix:738]: Hopefully it is a simplification across the board :)

**Comment by jaschdoc** [comment]
Is this what you had in mind?

**Pull request merged**